.pipelines/02-processing-data.yml

```diff
@@ -51,44 +51,7 @@ stages:
           python -m ml_service.pipelines.build_data_processing_pipeline
       displayName: 'Publish Data Preprocessing Pipeline'
 
-- stage: 'Trigger_Preprocessing_Pipeline'
-  displayName: 'Preprocess data'
-  condition: succeeded()
-  variables:
-    BUILD_URI: '$(SYSTEM.COLLECTIONURI)$(SYSTEM.TEAMPROJECT)/_build/results?buildId=$(BUILD.BUILDID)'
-  jobs:
-  - job: "Get_Preprocessing_Pipeline_ID"
-    condition: and(succeeded(), eq(coalesce(variables['auto-preprocess-data'], 'true'), 'true'))
-    displayName: "Get Preprocessing Pipeline ID for execution"
-    container: mlops
-    timeoutInMinutes: 0
-    steps:
-    - task: AzureCLI@1
-      inputs:
-        azureSubscription: '$(WORKSPACE_SVC_CONNECTION)'
-        scriptLocation: inlineScript
-        workingDirectory: $(Build.SourcesDirectory)
-        inlineScript: |
-          set -e # fail on error
-          export SUBSCRIPTION_ID=$(az account show --query id -o tsv)
-          python -m ml_service.pipelines.run_data_processing_pipeline --output_pipeline_id_file "preprocessing_pipeline_id.txt" --skip_preprocessing_execution
```

Comment:
It felt that the `run_data_processing_pipeline` script can be reused to retrieve any AML pipeline id. I introduced a `aml pipeline name` parameter to it. It also feels that the overlap with `run_training_pipeline` is quite huge except for the parameter part. But especially in the AzDO pipelines they immediate start is not used at all.

I would prefer to create a script maybe called `get_aml_pipeline_id` and rip out the starting part at all. I think the `run_*_pipeline` scripts are for dev-tests.

Reply:
Yes, only getting the pipeline id is used when running in ADO pipeline mode. The actual work, for example, data processing or training, is done in the serverless ML task to allow longer running jobs.